internal/tools/firestore/firestorequerycollectionparameterizable/firestorequerycollectionparameterizable.go

```diff
@@ -0,0 +1,622 @@
+				// The processed field might be an array format [a b c] or a single value
+				trimmedProcessed := strings.TrimSpace(processed)
+				
+				// Check if it's in array format [a b c]
+				if strings.HasPrefix(trimmedProcessed, "[") && strings.HasSuffix(trimmedProcessed, "]") {
+					// Remove brackets and split by spaces
+					arrayContent := strings.TrimPrefix(trimmedProcessed, "[")
+					arrayContent = strings.TrimSuffix(arrayContent, "]")
+					fields := strings.Fields(arrayContent) // Fields splits by any whitespace
+					for _, f := range fields {
+						if f != "" {
+							selectFields = append(selectFields, f)
+						}
+					}
+				} else {
+					selectFields = append(selectFields, processed)
+				}
+			}
+		} else {
+			selectFields = append(selectFields, field)
+		}
+	}
+	
+	return selectFields, nil
+}
+
+// processOrderBy processes the orderBy configuration with parameter substitution
+func (t Tool) processOrderBy(params map[string]any) (*OrderByConfig, error) {
+	if t.OrderByTemplate == nil {
+		return nil, nil
+	}
+	
+	orderBy := &OrderByConfig{}
+	
+	// Process field
+	field, err := t.processOrderByTemplate("field", params)
+	if err != nil {
+		return nil, err
+	}
+	orderBy.Field = field
+	
+	// Process direction
+	direction, err := t.processOrderByTemplate("direction", params)
+	if err != nil {
+		return nil, err
+	}
+	orderBy.Direction = direction
+	
+	if orderBy.Field == "" {
+		return nil, nil
+	}
+	
+	return orderBy, nil
+}
+
+func (t Tool) processOrderByTemplate(key string, params map[string]any) (string, error) {
+	var processedValue string
+	if value, ok := t.OrderByTemplate[key].(string); ok {
+		// Check if it's a template
+		if strings.Contains(value, "{{") {
+			processed, err := t.processTemplate(fmt.Sprintf("orderBy%s",key), value, params)
+			if err != nil {
+				return "", fmt.Errorf(errOrderByParseFailed, err)
+			}
+			processedValue = processed
+		} else {
+			processedValue = value
+		}
+	}
+	return processedValue, nil
+}
+
+// processLimit processes the limit field with parameter substitution
+func (t Tool) processLimit(params map[string]any) (int, error) {
+	limit := defaultLimit
+	if t.LimitTemplate != "" {
+		var processedValue string
+		
+		// Check if it's a template
+		if strings.Contains(t.LimitTemplate, "{{") {
+			processed, err := t.processTemplate("limit", t.LimitTemplate, params)
+			if err != nil {
+				return 0, fmt.Errorf(errLimitParseFailed, t.LimitTemplate, err)
+			}
+			processedValue = processed
+		} else {
+			processedValue = t.LimitTemplate
+		}
+		
+		// Try to parse as integer
+		if processedValue != "" {
+			parsedLimit, err := strconv.Atoi(processedValue)
+			if err != nil {
+				return 0, fmt.Errorf(errLimitParseFailed, processedValue, err)
+			}
+			limit = parsedLimit
+		}
+	}
+	return limit, nil
+}
+
+// processAnalyzeQuery processes the analyzeQuery field with parameter substitution
+func (t Tool) processAnalyzeQuery(params map[string]any) (bool, error) {
```

Comment:
Should we allow this to be a template param? This seems like a dev should chose to have this true or false